giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/aspects/access/RestSecurityAspect.java

```diff
@@ -287,8 +288,40 @@ public Object checkImageAccess(ProceedingJoinPoint joinPoint, ImageAccessCheck c
         
         return joinPoint.proceed(arguments);
     }
-    
-    
+
+    @Around("within(edu.asu.diging.gilesecosystem.web.rest..*) && @annotation(check)")
+    public Object checkDocumentOwner(ProceedingJoinPoint joinPoint, DocumentOwnerCheck check) throws Throwable {
+
+        UserTokenObject userTokenObj = extractUserTokenInfo(joinPoint, check.github(), check.value());
+
+        User user = userTokenObj.user;
+        String token = userTokenObj.token;
+        String docId = userTokenObj.parameter;
+
+        if (user == null) {
+            throw new AspectMisconfigurationException("User object is missing in method.");
+        }
+
+        IDocument doc = filesManager.getDocument(docId);
+        if (doc == null) {
+            return new ResponseEntity<String>(HttpStatus.NOT_FOUND);
+        }
```

Comment:
the same thing is checked in the controller method. is this method checking anything else that the giles token check doesn't do? If not, remove this method and just use ```TokenCheck``` annotation like the other methods.